libraries/src/Mail/MailTemplate.php

```diff
@@ -485,27 +486,18 @@ public static function deleteTemplate($key)
 	 *
 	 * @since   4.0.0
 	 */
-	protected function setAttachmentName($file, $name)
+	protected function getAttachmentName($file, $name)
 	{
-		// If no name is given use the basename of the file to be attached
+		// If no name is given, do not process it further
 		if (!trim($name))
 		{
-			return basename($file);
+			return '';
 		}
 
-		// Get file extension of the file to be attached
-		$extensionFile = pathinfo($file, PATHINFO_EXTENSION);
-
-		// If the file has no extension there is nothing to do with the name
-		if (!$extensionFile)
-		{
-			return $name;
-		}
-
-		// Make sure to attach the file with the same extension as the attached file has (case-insensitive)
-		if (strtolower(pathinfo($name, PATHINFO_EXTENSION)) !== strtolower($extensionFile))
+		// The given name use a different file extension, something could not be right, ignore it
+		if (File::getExt($file) !== File::getExt($name))
```

Comment:
This here is definitely wrong because if the $name has no extension, which is the preferred way how it should be used, and the real file has an extension, then always the real file name will be used. This condition should fire only if the File::getExt($name) is not empty string or null or whatever File::getExt returns if the re is no extension.

Reply:
We could change here to `if (File::getExt($name))`, i.e. if the $name has an extension, we ignore the $name and use the real file name, and later below we can do a `return $name . '.' . $extensionFile` instead of `return $name;`.

That would be simple and clear.

What do you think?

Reply:
or we use stripExt on the $name in any case and then just append the one of the file in any case